lib/Mojo/IOLoop/Stream/Server.pm

```diff
@@ -0,0 +1,125 @@
+package Mojo::IOLoop::Stream::Server;
+use Mojo::Base 'Mojo::IOLoop::Stream';
+
+use Mojo::Server;
+use Mojo::Util 'term_escape';
+
+use constant DEBUG => $ENV{MOJO_SERVER_DEBUG} || 0;
+
+has app => sub { Mojo::Server->new->build_app('Mojo::HelloWorld') };
+
+sub new {
+  my $self = shift->SUPER::new(@_);
+  $self->on(read  => sub { shift->_read_content(shift) });
+  $self->on(close => sub { shift->_close });
+  return $self;
+}
+
+sub _build_tx {
+  my $self = shift;
+
+  my $tx = $self->app->build_tx;
+  $self->emit(build_tx => $tx);
+  return $tx;
+}
+
+sub _close {
+  delete($_[0]->{tx})->closed if $_[0]->{tx};
```

Comment:
The private property `$self->{tx}` is only initialized in subclasses. That breaks encapsulation.